src/display/editor/tools.js

```diff
@@ -782,17 +786,21 @@ class AnnotationEditorUIManager {
     highlightColors,
     enableHighlightFloatingButton,
     enableUpdatedAddImage,
+    enableNewAltTextWhenAddingImage,
     mlManager
   ) {
-    this._signal = this.#abortController.signal;
+    const signal = (this._signal = this.#abortController.signal);
     this.#container = container;
     this.#viewer = viewer;
     this.#altTextManager = altTextManager;
     this._eventBus = eventBus;
-    this._eventBus._on("editingaction", this.#boundOnEditingAction);
-    this._eventBus._on("pagechanging", this.#boundOnPageChanging);
-    this._eventBus._on("scalechanging", this.#boundOnScaleChanging);
-    this._eventBus._on("rotationchanging", this.#boundOnRotationChanging);
+    this._eventBus._on("editingaction", this.#boundOnEditingAction, { signal });
+    this._eventBus._on("pagechanging", this.#boundOnPageChanging, { signal });
+    this._eventBus._on("scalechanging", this.#boundOnScaleChanging, { signal });
+    this._eventBus._on("rotationchanging", this.#boundOnRotationChanging, {
+      signal,
+    });
+    this._eventBus._on("setpreference", this.#boundOnSetPreference);
```

Comment:
It seems that the "setpreference" listener is missing the signal-option!

Also, now that we're removing the listeners with an `AbortSignal`, we can probably define these inline instead?
```suggestion
    this._eventBus._on("editingaction", this.onEditingAction.bind(this), { signal });
    this._eventBus._on("pagechanging", this.onPageChanging.bind(this), { signal });
    this._eventBus._on("scalechanging", this.onScaleChanging.bind(this), { signal });
    this._eventBus._on("rotationchanging", this.onRotationChanging.bind(this), {
      signal,
    });
    this._eventBus._on("setpreference", this.onSetPreference.bind(this), { signal });
```